packages/server/graphql/mutations/loginWithGoogle.ts

```diff
@@ -86,22 +87,28 @@ const loginWithGoogle = {
         }
       }
 
+      const userId = `google-oauth2|${sub}`
+      const deletedUser = await r
+        .table('User')
+        .get(userId)
+        .run()
+
       // it's a new user!
       const nickname = name || email.substring(0, email.indexOf('@'))
       const preferredName = nickname.length === 1 ? nickname.repeat(2) : nickname
-      const userId = `google-oauth2|${sub}`
       const identity = new AuthIdentityGoogle({
         id: sub,
         isEmailVerified: email_verified !== 'false'
       })
       const newUser = new User({
-        id: userId,
         preferredName,
         picture,
         email,
         identities: [identity],
         segmentId
       })
+      // if a deleted user with the same id exists, generate a new id
+      newUser.id = deletedUser && deletedUser.isRemoved ? `google-oauth2|${generateUID()}` : userId
```

Comment:
nit: instantiate the newUser object with the correct ID instead of mutating the object down below